src/app/seller-post-ad-form/seller-post-ad-form.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, OnInit, ɵConsole} from '@angular/core';
+import { ImageProperty } from './imageProperty';
+import { DatePipe } from '@angular/common';
+import {NgForm} from '@angular/forms';
+
+@Component({
+  selector: 'seller-post-ad-form',
+  templateUrl: './seller-post-ad-form.component.html',
+  styleUrls: ['./seller-post-ad-form.component.css']
+})
+
+export class SellerPostAdFormComponent implements OnInit {
+  minNoOfImage=1;
+  maxNoOfImage=5;
+  isAddressSelected:boolean=false;
+  heroImageUrl = ""; // store url of hero image
+  imageUrlArray = []; // store url of images
+  imageArray:ImageProperty[] =[];
+  
+  categories = ["Home","Electronics","Car","Bike"]; // this is provided by categories api
+  states = ["Andra Pradesh","Go","Gujarat","Haryana","Himachal Pradesh","Jammu and Kashmir","Jharkhand","Karnataka",
+  "Kerala","Madya Pradesh","Maharashtra","Punjab","Rajasthan"]
+  
+  //properties of html element 
+  addressDisplayValue = "none";
```

Comment:
you can handle this using isAddressSelected itself
Not need to add extra variable